framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -129,6 +134,20 @@ public Void visitAnnotation(AnnotationTree node, Void p) {
     }
 
     @Override
+    protected void commonAssignmentCheck(
+            AnnotatedTypeMirror varType,
+            ExpressionTree valueExp,
+            /*@CompilerMessageKey*/ String errorKey) {
+
+        // This is definitely a hack. However, it does prevent some spurious warnings
+        // from the Value Checker when interacting with unannotated bytecode.
+        if (varType.toString().contains("@")
```

Comment:
Oh, the second condition is different because it is checking if the underlying (Java) type has an annotation.  That happens when the type has an explicit annotation.  I would guess that it prevents test failures by chance.  Does that make sense?